PR: https://github.com/odoo/odoo/pull/257484

From: e7407b618cccf58183277c64e0c7648ddaa29b12
From: kcv-odoo
Date: 2026-09-09 06:21:48

Breaking data model changes scores: del:2 + add:0, change matches:
-    _inherit = 'res.country'
-    _inherit = 'res.partner'

Total Changes: 57

[REF] base,portal{,address_extended},*: refactor portal address

This commit improve portal address layout and refactor address logic:

* Improve countrty's form view to make title bigger.
* Make zip_required (boolean) field a selection field zip_applicability
with not_applicable, optional and required values and update countries
data, now in portal page zip will only be visible if country does not
have zip not_applcable.
* Merge `portal_address_extended` module in portal module
to make it easier to maintain by making `portal` module dependent
on `base_address_extended`.
* Improve portal address layout and adapt localisation.
* Also updated base_address_extended dependency to make
it independent of contacts app and move `Cities` menu in
contacts app and make it dependent on `base_address_extended`.

task-5208254

closes odoo/odoo#257484

Related: odoo/enterprise#112947
Related: odoo/upgrade#10598
Signed-off-by: Victor Feyens (vfe) <vfe@odoo.com>

================================= pseudo patch: =================================

--- a/addons/portal_address_extended/models/__init__.py
+++ b/None
@@ -1,3 +0,0 @@
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from . import res_country, res_partner

--- a/addons/portal_address_extended/models/res_country.py
+++ b/None
@@ -1,34 +0,0 @@
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from odoo import models
-
-
-class ResCountry(models.Model):
-    _inherit = 'res.country'
-
-    def _enforce_city_choice(self):
-        if not self:
-            return False
-
-        # Only enabled on frontend for those countries for now
-        # Feature has to be adapted to be more generic and less blocking
-        # before being enabled for other countries
-        if self.code not in ['BR', 'CL', 'PE', 'CO', 'TW']:
-            return False
-
-        self.ensure_one()
-        return self.enforce_cities and bool(
-            self.env['res.city'].sudo().search_count([('country_id', '=', self.id)], limit=1)
-        )
-
-    def _get_cities(self, state_id=None):
-        if not self:
-            return self.env['res.city'].sudo()
-
-        return self.env['res.city'].sudo().search([
-            ('country_id', '=', self.id),
-            ('state_id', 'in', [state_id, False]),
-        ])
-
-    def _get_cities_fields_to_fetch(self):
-        return ['id', 'name', 'state_id']

--- a/addons/portal_address_extended/models/res_partner.py
+++ b/None
@@ -1,20 +0,0 @@
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from odoo import models
-
-
-class ResPartner(models.Model):
-    _inherit = 'res.partner'
-
-    def _get_frontend_writable_fields(self):
-        frontend_writable_fields = super()._get_frontend_writable_fields()
-        frontend_writable_fields.add('city_id')
-
-        return frontend_writable_fields
-
-    def _get_mandatory_address_fields(self, country_sudo, **kwargs):
-        mandatory_fields = super()._get_mandatory_address_fields(country_sudo, **kwargs)
-        if country_sudo._enforce_city_choice():
-            mandatory_fields.add('city_id')
-            mandatory_fields.remove('city')
-        return mandatory_fields
